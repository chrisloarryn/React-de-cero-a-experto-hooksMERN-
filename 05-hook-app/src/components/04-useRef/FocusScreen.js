import React, { useRef } from 'react'
import '../02-useEffect/effects.css'

export const FocusScreen = () => {
  const inputNameRef = useRef()
  const inputAgeRef = useRef()

  const handleClick = e => {
    e.preventDefault()
    inputNameRef.current.select()
    // document.querySelector('input').focus()
  }
  return (
    <form>
      <h1>Focus Screen</h1>
      <div className="form-group">
        <label htmlFor="exampleInputName">Your name:</label>
        <input
          ref={inputNameRef}
          type="text"
          className="form-control"
          id="exampleInputName"
          aria-describedby="nameHelp"
        />
        <small id="nameHelp" className="form-text text-muted">
          We'll never share your name with anyone else.
        </small>
      </div>
      <div className="form-group">
        <label htmlFor="exampleInputAge">Your age:</label>
        <input
          ref={inputAgeRef}
          type="text"
          className="form-control"
          id="exampleInputAge"
          aria-describedby="ageHelp"
        />
        <small id="ageHelp" className="form-text text-muted">
          We'll never share your age with anyone else.
        </small>
      </div>

      <button onClick={handleClick} className="btn btn-outline-primary mt-1">
        Focus
      </button>
    </form>
  )
}
