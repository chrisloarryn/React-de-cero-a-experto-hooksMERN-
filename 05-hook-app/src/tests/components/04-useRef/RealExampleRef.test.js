import React from 'react'
import { shallow } from 'enzyme'
import { RealExampleRef } from './../../../components/04-useRef/RealExampleRef'

describe('Tests in <RealExampleRef />', () => {
  const wrapper = shallow(<RealExampleRef />)

  test('Should be rendered correctly', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false)
  })
  test('Should render <MultipleCustomHooks />', () => {
    wrapper.find('button').simulate('click')
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true)
  })
})
