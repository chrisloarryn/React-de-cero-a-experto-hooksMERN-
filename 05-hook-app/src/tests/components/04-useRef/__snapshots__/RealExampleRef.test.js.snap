// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <RealExampleRef /> Should be rendered correctly 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-info mt-2"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
